src/main/java/org/apache/rocketmq/flink/legacy/RocketMQSourceFunction.java

```diff
@@ -241,91 +243,58 @@ public void run(SourceContext context) throws Exception {
                                     () -> {
                                         while (runningChecker.isRunning()) {
                                             try {
-                                                long offset = getMessageQueueOffset(mq);
-
-                                                PullResult pullResult = null;
+                                                getMessageQueueOffset(mq);
                                                 if (StringUtils.isEmpty(sql)) {
-                                                    pullResult =
-                                                            consumer.pullBlockIfNotFound(
-                                                                    mq, tag, offset, pullBatchSize);
+                                                    consumer.subscribe(topic, tag);
```

Comment:
this can be put out of the while loop

Reply:
I agree with @keranbingaa. The `consumer.subscribe(topic, tag)` could be move to the while loop.

Reply:
ok